File: python/7 kyu/166.py
# https://www.codewars.com/kata/5ce6cf94cb83dc0020da1929/train/python

# Passed

def uglify_word(s):
    flag = 1
    result = str()
    for letter in s:
        if letter.isalpha():
            if flag == 1:
                result += letter.upper()
            elif flag == 0:
                result += letter.lower()        
            
            flag = int(not flag)
        else:
            result += letter
            flag = 1
            
    return result
        
output = uglify_word("qwe123asdf456zxc")
print(output)

File: nodejs/7 kyu/166.js
// https://www.codewars.com/kata/5b4e474305f04bea11000148/train/javascript

// Passed

function digits(num) {
  let allDigits = String(num)
    .split("")
    .map((digit) => Number(digit));
  let allSums = [];
  let currentIdx = 0;
  for (const firstDigit of allDigits) {
    for (const secondDigit of allDigits.slice(currentIdx + 1)) {
      let sum = firstDigit + secondDigit;
      allSums.push(sum);
    }

    currentIdx++;
  }

  return allSums;
}

const output = digits(81596);
console.log(output);